classes/allocation_analyzer.py

```diff
@@ -0,0 +1,126 @@
+from csv import DictReader
+from collections import (
+    defaultdict,
+    namedtuple,
+)
+from classes.judge import Judge
+from classes.startup import Startup
+Assignment = namedtuple("Assignment", ["judge", "startup"])
+
+
+def increment_assignment_count(startup, metric_name, analysis):
+    name = startup['name']
+    analysis[name][metric_name] += 1
+
+
+def female_judge(assignment, read_counts):
+    judge, startup = assignment
+    if judge.is_female():
+        increment_assignment_count(startup, 'female_reads', read_counts)
+
+
+def role_distribution(assignment, read_counts):
+    judge, startup = assignment
+    metric_name = "%s_reads" % judge['role']
+    increment_assignment_count(startup, metric_name, read_counts)
+
+
+def program_match(assignment, read_counts):
+    judge, startup = assignment
+    metric_name = "home_program_reads"
+    if startup['program'] == judge['program']:
+        increment_assignment_count(startup, metric_name, read_counts)
+
+
+def industry_match(assignment, read_counts):
```

Comment:
The parallels with program_match strongly suggests that this should be refactored.  Ideally, there should be no feature specific code, but it should all be inferred from the features and values that get used in the input files.